### ISSUE TITLE
When in remote repositories (Internet archive, ect built in), the screen flashes

### ISSUE THREAD

**Issue by 100Create** [johnfactotum/foliate#1361]
When in remote repositories (Internet archive, ect built in), the screen flashes

**To Reproduce**
1. Go to 'Internet Archive' (or any remote repo)
2. Click on anything that would display book covers
3. See flashing error

**Expected behavior**
I would expect the entire application not to flash and work normally

**Version:**
System: Ubuntu 24.04 LTS (Noble Numbat)
Desktop: ubuntu:GNOME
Session: ubuntu (x11)
Language: en_US.UTF-8

Versions:
- Foliate 3.1.1
- GJS 1.80.02
- GTK 4.14.2
- Adwaita 1.5.0
- GLib 2.80.0
- WebKitGTK 2.44.2

User directories:
- /home/user/.local/share
- /home/user/.cache

**Comment by kamihatsu** [comment]
Hello, what video chip do u have?

**Comment by kamihatsu** [comment]
On Fredora 40 KDE Plasma its  work ok without flashes, bug isnt confirmed

**Comment by 100Create** [comment]
> Hello, what video chip do u have?

Sorry for the late reply, I am using an Nvidia RTX 3060 Laptop GPU on GNOME with a 12th gen Intel Core i7-12700H, it now crashes upon attempting to do anything in the app, I have reinstalled foliate from apt a few times now (I don't want the other versions as I want easily changable themes and the other gnome standards)

**Comment by 100Create** [comment]
>Fredora 40 KDE Plasma its work ok without flashes, bug isnt confirmed

You are on a different desktop environment entirely, please don't say "It works fine on my machine" because your machine is not my machine.

**Comment by johnfactotum** [comment]
Yeah, a video showing the issue would be nice. But from your description, it seems like a WebKitGTK/Nvidia bug.

**Comment by o1Suleyman** [comment]
does this occur on Wayland?

**Comment by 100Create** [comment]
![image](https://github.com/user-attachments/assets/741fbb15-b7d4-4a8c-975b-c30aac372ea4)
Honestly whatever rendering is being used is just not very good. I am on X11 and cannot be bothered to switch to wayland. (THIS IS A NEW ISSUE)

**Comment by 100Create** [comment]
![image](https://github.com/user-attachments/assets/0b5b27d4-8628-4eb9-bd11-40ca843ab284)

**Comment by johnfactotum** [comment]
That kind of looks like the same issue as #1412.

**Comment by govarthenan** [comment]
This is not a driver bug. I believe the cause is in Foliate.

`src/library.js` gives the catalogue WebView a fully transparent background. With a transparent background, WebKitGTK composites each new frame over the previous one instead of clearing the damaged area first, so the old pixels stay visible and the view smears.

`src/selection-tools.js:337` makes the same call, which is why #1412 shows the same effect in the dictionary/translation popover. These two WebViews are the only transparent ones in Foliate, and they are exactly the two views in the reports. The reader WebView is opaque and has no reports.

The GPU path is not involved (each one verified in `/proc/<pid>/environ` of the `WebKitWebProcess`, all still smear): `WEBKIT_DISABLE_DMABUF_RENDERER=1`, `WEBKIT_DISABLE_COMPOSITING_MODE=1`, `WEBKIT_SKIA_ENABLE_CPU_RENDERING=1`, `GSK_RENDERER=cairo`, `GSK_RENDERER=gl`, `GSK_RENDERER=vulkan`.

A fix I found: Painting an opaque background removes the smear, but on its own it also changes the colour of the catalogue area from the window colour to WebKit's own base colour. Taking the colour from `window_bg_color` and injecting it into the page as well keeps the present look exactly:

Tested against `gtk4` HEAD (67b6676) on Linux Mint 22.3, Cinnamon, X11, GTK 4.14.5, WebKitGTK 2.52.3, NVIDIA 595.84. The smear is gone on load and on scroll, and the background colour is unchanged.

Happy to open a PR @johnfactotum , and a separate one for `selection-tools.js` (#1412).
